### DISCUSSION TITLE
Can I swap cookie session to bearer token?

### DISCUSSION THREAD

**Discussion by henrik-funmed** [ory/kratos#1988] [Q&A] [answered]
After using Kratos to log in a user with social login the credentials are set as cookies. I then find examples of how to set up Oathkeeper to check cookie credentials to protect endpoints.

Is there a way to configure Oathkeeper/Kratos so that I, after above login, can convert the received cookie session to bearer token instead when accessing the protected endpoints?

Like this:
1. Login using social login
2. Access API using bearer token

**Comment by vinckr** [comment]
Hello @henrik-funmed,
CHeck out the `id-token` mutator: https://www.ory.sh/oathkeeper/docs/pipeline/mutator/#id_token.

**Reply by popstr** [reply]
That could be handy, but don't you have to verify the session token in oathkeeper anyway (using the `check_session_url`)?

**Comment by popstr** [comment]
I am working with @henrik-funmed on this.

The problem is when we try to validate the session through the url given at `check_session_url` in our config (which is set to http://[kratos]/sessions/whoami) Kratos can't see the session token. Kratos seems to require the token to be sent as a cookie, but we would like to use a header with a Bearer token. Oathkeeper can easily be configured to use a bearer token, but it seems that Kratos can not.

Of course we could point Oathkeeper (using `check_session_url`) to a new proxy service that takes a request with an Authorization header, transforms it into a request with a cookie, validates the session against Kratos and the returns the data, but it shouldn't be necessary to implement a separate service.

**Comment by popstr** [comment, marked as answer]
It seems like the feature we are after is implemented in the latest version of Kratos! 👍

**Reply by vinckr** [reply]
Just for clarification and others who might have the same question: 
Do you mean the "[Ory Session Token](https://www.ory.sh/kratos/docs/concepts/session#ory-session-token)" ?

**Reply by popstr** [reply]
Yes, it is the Ory Session Token that we want Kratos to read from a header (instead of a cookie). In the latest version of the code Kratos checks for a Bearer token in the `Authorization` header, which is precisely what we want. It also checks headers called `X-Session-Token` and `X-Session-Cookie`.
It would be nice if this was configurable instead of Kratos guessing where the token might be, but I'm happy with the current solution.

**Reply by vinckr** [reply]
Thanks for clarifying.
If it would be useful to have and you have an idea or proposal for making this configurable feel free to open a [feature request](https://github.com/ory/kratos/issues/new?assignees=&labels=feat&template=FEATURE-REQUEST.yml), maybe there are others in the community who would like this implemented.
